bundle/src/main/java/com/adobe/acs/commons/wcm/impl/SiteMapServlet.java

```diff
@@ -274,18 +285,33 @@ private void writeAssets(final XMLStreamWriter stream, final Resource assetFolde
         }
     }
 
-    private void writeFirstPropertyValue(final XMLStreamWriter stream, final String elementName, final String[] propertyNames,
-                                         final ValueMap properties) throws XMLStreamException {
+    private void writeFirstPropertyValue(final XMLStreamWriter stream, final String elementName,
+            final String[] propertyNames, final ValueMap properties) throws XMLStreamException {
+        for (String prop : propertyNames) {
+            String value = properties.get(prop, String.class);
+            if (value != null) {
+                writeElement(stream, elementName, value);
+                break;
+            }
+        }
+    }
+
+    private void writeFirstPropertyValue(final XMLStreamWriter stream, final String elementName,
+            final String[] propertyNames, final InheritanceValueMap properties) throws XMLStreamException {
         for (String prop : propertyNames) {
             String value = properties.get(prop, String.class);
```

Comment:
Not a big deal, but this line can just be `value = properties.getInherited(prop, String.class)` as `getInherited(..)` tries to get the the local property first, and only traverses up if it cannot be found locally.

https://docs.adobe.com/content/docs/en/aem/6-2/develop/ref/javadoc/com/day/cq/commons/inherit/InheritanceValueMap.html#getInherited(java.lang.String,%20java.lang.Class)